### ISSUE TITLE
NullPointer on from parameter with negative value and 500 code instead of 400 error

### ISSUE THREAD

**Issue by gilfernandes** [elastic/elasticsearch#23324]
I have tried to fire a silly request to ElastiSearch 5.1.1:

GET products/_search
```json
{
  **"from": -2**,
  "size": 100, 
  "query": {
    "match": {
      "productName": "soy sauce"
    }
  },
  "sort": [
    {
      "customerRating": {
        "order": "desc"
      }
    },
    "_score",
    {
      "price" : {
        "order": "asc"
      }
    }
  ]
}
```

and got this:

```json
{
  "error": {
    "root_cause": [],
    "type": "reduce_search_phase_exception",
    "reason": "[reduce] ",
    "phase": "query",
    "grouped": true,
    "failed_shards": [],
    "caused_by": {
      "type": "null_pointer_exception",
      "reason": null
    }
  },
  "status": 503
}
```

I had expected a 400 (user error) and a meaningful error message, like e.g: from parameter cannot be negative.

**Comment by cbuescher** [comment]
I agree we should catch this earlier. I just reproduced this on 5.1.2. and there I get an ArrayIndexOutOfBoundsException from
```
java.lang.ArrayIndexOutOfBoundsException: -2
	at org.elasticsearch.action.search.SearchPhaseController.sortDocs(SearchPhaseController.java:214) ~[elasticsearch-5.1.2.jar:5.1.2]
	at org.elasticsearch.action.search.SearchQueryThenFetchAsyncAction.moveToSecondPhase(SearchQueryThenFetchAsyncAction.java:80) ~[elasticsearch-5.1.2.jar:5.1.2]
```

I think we should reject negative `from` values already in SearchSourceBuilder, which would mean that errors would be triggered either on the client side or the coordinating node.

**Comment by cbuescher** [comment]
Closed by #23358
